spec/tools/environment_builders/openstack/services/compute/builder.rb

```diff
@@ -64,9 +64,13 @@ def find_or_create_servers(volumes, networks, security_groups, images, networkin
                             }) if volume
             end
 
-            if (network_name = server.delete(:__network_name))
-              network = networks.detect { |x| x.name == network_name }
-              server.merge!(:nics => [{"net_id" => network.id}]) if network
+            if (network_names = server.delete(:__network_names))
+              nics = []
+              network_names.each do |network_name|
+                network = networks.detect { |x| x.name == network_name }
+                nics << {"net_id" => network.id} if network
+              end
+              server.merge!(:nics => nics) if nics
```

Comment:
`server[:nics] = nics if nics` has the same effect but it's a bit simpler and works faster.

```ruby
Benchmark.ips do |x|
  x.report('merge!') { {}.merge!(a: 1) }
  x.report('[]=') { {}[:a] = 1 }
  x.compare!
end
__END__
Calculating -------------------------------------
              merge!    22.702k i/100ms
                 []=    34.332k i/100ms
-------------------------------------------------
              merge!    381.508k (± 7.9%) i/s -      1.907M
                 []=    730.552k (± 6.1%) i/s -      3.639M

Comparison:
                 []=:   730552.0 i/s
              merge!:   381507.6 i/s - 1.91x slower
```